client/pages/DeckStat.jsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Col, Row } from 'react-bootstrap';
+import { useTranslation } from 'react-i18next';
+import Panel from '../Components/Site/Panel';
+import CardBack from '../Components/Decks/CardBack';
+import { loadStats } from '../redux/actions';
+import { clearApiStatus } from '../redux/actions';
+
+import { BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, Legend, Scatter,Radar, RadarChart, PolarGrid, PolarAngleAxis, PolarRadiusAxis, } from 'recharts';
+// const data = [
+//     {
+//         level: 'Beginner',
+//         win: 40,
+//         loss: 2,
+//         ratio: 89
+//     },
+//     {
+//         level: 'Casual',
+//         win: 40,
+//         loss: 40,
+//         ratio: 50
+//     },
+//     {
+//         level: 'Competitive',
+//         win: 20,
+//         loss: 80,
+//         ratio: 20
+//     }
+// ];
+
+// const renderCustomizedLabel = (props) => {
+//     const { x, y, width, value } = props;
+//     const radius = 10;
+//     return (
+//       <g>
+//         <circle cx={x + width / 2} cy={y - radius} r={radius} fill="#8884d8" />
+//         <text x={x + width / 2} y={y - radius} fill="#fff" textAnchor="middle" dominantBaseline="middle">
+//           {value.split(' ')[1]}
+//         </text>
+//       </g>
+//     );
+//   };
+const DecksComponent = () => {
+    const { user } = useSelector((state) => ({
+        user: state.account.user
+    }));
+    const { deck } = useSelector((state) => ({
+        deck: state.cards.selectedDeck
+    }));
+    let { stats } = useSelector((state) => ({
+        stats: state.stats
+    }));
+    const dispatch = useDispatch();
+    const { t } = useTranslation();
+
+    useEffect(() => {
+            // if (user != null && deck != null) {
+            dispatch(
+                loadStats({
+                    userId: 1,
+                    deckId: 5
```

Comment:
1rst promblem,
so here im supposed to use user.id and deck.id, the one loaded at line 45, 48, the value of user and deck are nul here but not at line 73,74

Reply:
You're in a useEffect callback. This is called whenever the things in the array change. So you want deck.id and user.id to be in the array. Then only set these values if they are present

Reply:
So this problem is now fix, thanks